File: projetmanga-server/models/ReadingHistory.js
const mongoose = require('mongoose');

const ReadingHistorySchema = new mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true,
        index: true
    },
    mangaId: {
        type: String,
        required: true
    },
    source: {
        type: String,
        enum: ['local', 'jikan'],
        required: true
    },
    currentChapterId: {
        type: String,
        required: false
    },
    currentChapterNumber: {
        type: Number,
        required: false
    },
    lastReadAt: {
        type: Date,
        default: Date.now
    },
    title: String,
    imageUrl: String
}, { timestamps: true });

ReadingHistorySchema.index({ user: 1, mangaId: 1 }, { unique: true });

module.exports = mongoose.model('ReadingHistory', ReadingHistorySchema);

File: projetmanga-client-android/app/src/main/java/com/mangaproject/data/model/ReadingHistory.kt
package com.mangaproject.data.model

import com.google.gson.annotations.SerializedName

data class ReadingHistory(
    @SerializedName("_id")
    val id: String,
    @SerializedName("mangaId")
    val mangaId: String,
    @SerializedName("source")
    val source: String, // "local" ou "jikan"
    @SerializedName("currentChapterId")
    val currentChapterId: String?,
    @SerializedName("currentChapterNumber")
    val currentChapterNumber: Int?,
    @SerializedName("lastReadAt")
    val lastReadAt: String,
    @SerializedName("title")
    val title: String?,
    @SerializedName("imageUrl")
    val imageUrl: String?
)

data class UpdateReadingHistoryRequest(
    val mangaId: String,
    val source: String,
    val chapterId: String?,
    val chapterNumber: Int?,
    val title: String?,
    val imageUrl: String?
)

data class ReadingHistoryResponse(
    val history: List<ReadingHistory>
)

data class SingleReadingHistoryResponse(
    val history: ReadingHistory?
)

